src/js/legend.js

```diff
@@ -0,0 +1,228 @@
+/*!
+Copyright 2015 OCAD University
+
+Licensed under the New BSD license. You may not use this file except in
+compliance with this License.
+
+You may obtain a copy of the License at
+https://github.com/gpii/universal/LICENSE.txt
+*/
+
+(function ($, fluid) {
+
+    "use strict";
+
+    fluid.defaults("gpii.chartAuthoring.pieChart.legend", {
+        gradeNames: ["gpii.d3ViewComponent", "autoInit"],
```

Comment:
This is a good question. Looking at the legend implementation, using html should be fine too. @waharnum, do you have some insights?

With some more research, D3 provides a d3.legend (http://bl.ocks.org/ZJONSSON/3918369), for creating a legend for the corresponding chart. The generated legend even tries to match colors with what are used in the chart. This reminds me another issue with the current implementation that both pie.js and legend.js generate their own color arrays which could cause an out-of-sync problem. The color generation should be dealt with at a high level like gpii.chartAuthoring.pieChart component then being distributed to children components.